### PR TITLE
DataFrames support

### PR DESCRIPTION
**Pull request by srgg**
## READY for Review

### CODE REVIEW

File: connector/src/test/scala/com/basho/riak/spark/rdd/TimeSeriesTest.scala [hunk 1]
```diff
@@ -160,4 +168,55 @@ class TimeSeriesTest extends AbstractRDDTest {
...
+            StructField("user_id", StringType, nullable=false),
+            StructField("temperature_k", DoubleType, nullable=true)))
+      )
+      .load(DEFAULT_TS_NAMESPACE.getBucketTypeAsString)
+      .filter(s"time > CAST($from AS TIMESTAMP) AND time < CAST($to AS TIMESTAMP) AND surrogate_key = 1 AND family = 'f'")
+
+    df.printSchema()
+    val data = df.toJSON.collect()
+
+    assertEqualsUsingJSONIgnoreOrder("[" +
```
Comment (zkhadikova) [inline connector/src/test/scala/com/basho/riak/spark/rdd/TimeSeriesTest.scala]: Timestamps when converted to strings may have different values depending on timezone, which makes test fail.

Comment (srgg) [inline connector/src/test/scala/com/basho/riak/spark/rdd/TimeSeriesTest.scala]: Good catch, I'll re-check this

File: connector/src/main/scala/org/apache/spark/sql/riak/RiakRelation.scala [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  }
+
+  private[this] val baseRdd: RiakTSRDD[Row] =
+    sqlContext.sparkContext.riakTSBucket[Row](bucket)
+
+  def buildScan(): RDD[Row] = baseRdd.asInstanceOf[RDD[Row]]
+
+  override def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row] = {
+
+    val prunedRdd = requiredColumns.isEmpty match {
```
Comment (zkhadikova) [inline connector/src/main/scala/org/apache/spark/sql/riak/RiakRelation.scala]: It's better to use if else construct for single boolean.

Comment (srgg) [inline connector/src/main/scala/org/apache/spark/sql/riak/RiakRelation.scala]: but how we can initialize val in that case?

**Comment by tmatvienko** [comment]
Reviewed, ready for merge

**Comment by zkhadikova** [comment]
+1

**Pull request merged**